analyses/fusion-frequencies/01-fusion-frequencies.R

```diff
@@ -0,0 +1,296 @@
+                Kids_First_Participant_ID, 
+                FusionName, 
+                Gene1A, 
+                Gene1B, 
+                Gene2A, 
+                Gene2B, 
+                Fusion_Type,
+                DomainRetainedGene1A,
+                DomainRetainedGene1B,
+                reciprocal_exists,
+                ends_with("anno")) %>%
+  mutate(Alt_ID=paste(FusionName, Fusion_Type),sep="_") %>%
+  rename("Kids_First_Biospecimen_ID"="Sample",
+         "Kinase_domain_retained_Gene1A" = "DomainRetainedGene1A",
+         "Kinase_domain_retained_Gene1B" = "DomainRetainedGene1B",
+         "Reciprocal_exists" = "reciprocal_exists")%>%
+  # replace NA to "" in columns that have NA
+  replace_na(list("Kinase_domain_retained_Gene1A"="",
+                  "Kinase_domain_retained_Gene1B"="",
+                  "Gene1A_anno"="",
+                  "Gene1B_anno"="",
+                  "Gene2A_anno"="",
+                  "Gene2B_anno"="",
+                  "Fusion_anno"=""))%>%
+  # We want a single column that contains the gene symbols
+  tidyr::gather(Gene1A, Gene1B, Gene2A, Gene2B,
+                key = gene_position, value = gene_symbol) %>%
+  filter(!is.na(gene_symbol))
+
+
+rm(tumor_kfbids)
+suppressMessages(gc(reset = TRUE))
+
+
+
+# Compute mutation frequencies -------------------------------------------------
+message('Compute mutation frequencies...')
+cancer_group_cohort_summary_df <- get_cg_cs_tbl(td_htl_dfs$overall_htl_df)
+
+# nf = n_samples filtered
+nf_cancer_group_cohort_summary_df <- cancer_group_cohort_summary_df %>%
+  filter(n_samples >= 5)
+
+# Run get_cg_ch_mut_freq_tbl() per cancer_group and cohort
+fus_freq_tbl_list <- lapply(
+  1:nrow(nf_cancer_group_cohort_summary_df),
+  function(i) {
+    cgcs_row <- nf_cancer_group_cohort_summary_df[i, ]
+    stopifnot(identical(nrow(cgcs_row), as.integer(1)))
+    stopifnot(identical(length(cgcs_row$cohort_list), as.integer(1)))
+    stopifnot(is.character(cgcs_row$cohort_list[[1]]))
+    
+    c_cancer_group <- cgcs_row$cancer_group
+    c_cohorts <- cgcs_row$cohort_list[[1]]
+    stopifnot(identical(paste(c_cohorts, collapse = '&'), cgcs_row$cohort))
+    message(paste(c_cancer_group, cgcs_row$cohort))
+    
+    # Call function for each cohort and cancer_group
+    res_df <- get_cg_ch_mut_freq_tbl(
+      fusion_df, td_htl_dfs$overall_htl_df, td_htl_dfs$primary_htl_df,
+      td_htl_dfs$relapse_htl_df, c_cancer_group, c_cohorts) %>%
+      # if Alt_ID has no matched Patients_in_dataset
+      filter(!is.na(Patients_in_dataset))
+    
+    if ( nrow(res_df) != 0 ){
+    # merge fusion dataframe with the counts and frequencies in each cancer_group
+    # - per cohort 
+    # - per primary samples in cohort
+    # - per relapse samples in cohort
+    fusion_df <- fusion_df %>% 
+      select(-Kids_First_Biospecimen_ID,-Kids_First_Participant_ID) %>%
+      unique() %>%
+      inner_join(res_df,by="Alt_ID") 
+    
+    return(fusion_df)
+    }
```

Comment:
```suggestion
    if ( nrow(res_df) != 0 ){
      # merge fusion dataframe with the counts and frequencies in each cancer_group
      # - per cohort 
      # - per primary samples in cohort
      # - per relapse samples in cohort
      fusion_df <- fusion_df %>% 
        select(-Kids_First_Biospecimen_ID,-Kids_First_Participant_ID) %>%
        unique() %>%
        inner_join(res_df,by="Alt_ID") 

      return(fusion_df)
    }
```

When `nrow(res_df) == 0`, it seems that `NULL` will be returned, will the following code still be able to run correctly?

Reply:
since it's NULL it doesn't have anything to bind so it seems to work I can check again.